thunder/tests/opinfos.py

```diff
@@ -2288,6 +2289,38 @@ def fmod_sample_input_generator(op, device, dtype, requires_grad, **kwargs):
 )
 elementwise_binary_ops.append(logical_xor_opinfo)
 
+
+def ldexp_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    a = make((4, 4), **kwargs)
+    b = make((4, 4), **kwargs)
+    c = make((4, 1), **kwargs)
+
+    yield SampleInput(a, b)
+    yield SampleInput(a, c)
+
+
+def ldexp_error_generator(op, device, dtype=torch.float32, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype)
+
+    if torch.cuda.is_available():
+        a = make((4, 4), device="cuda")
+        b = make((4, 4), device="cpu")
+
+        err_msg = "Expected all tensors to be on the same device, but found at least two devices, cuda and cpu"
+        yield (SampleInput(a, b), RuntimeError, err_msg)
+
+
+ldexp_opinfo = OpInfo(
+    ltorch.ldexp,
+    supports_grad=True,
+    dtypes=(datatypes.float32, datatypes.float64),
```

Comment:
Can this op be defined for all floating types or all inexact types instead of just fp32 and fp64?

Reply:
Nice catch. updated the tests to `datatypes.all_dtypes`.

Thank you!